test/truefi2/lines-of-credit/TrueFiCreditOracle.test.ts

```diff
@@ -27,137 +31,203 @@ describe('TrueFiCreditOracle', () => {
+    })
+
+    it('max borrower limit is properly set', async () => {
+      await oracle.connect(manager).setMaxBorrowerLimit(borrower.address, 1_000_000)
+      expect(await oracle.getMaxBorrowerLimit(borrower.address)).to.equal(1_000_000)
     })
 
-    it('max borrower limit is set correctly for account', async () => {
-      expect(await oracle.getMaxBorrowerLimit(firstAccount.address)).to.equal(firstBorrowLimit)
+    it('updates eligible until time', async () => {
+      const tx = await oracle.connect(manager).setMaxBorrowerLimit(borrower.address, 1_000_000)
+      const timestamp = BigNumber.from((await provider.getBlock(tx.blockNumber)).timestamp)
+      expect(await oracle.eligibleUntilTime(borrower.address)).to.equal(timestamp.add(31 * DAY))
     })
 
-    it('change existing max borrower limit', async () => {
-      await oracle.connect(manager).setMaxBorrowerLimit(firstAccount.address, secondBorrowLimit)
-      expect(await oracle.getMaxBorrowerLimit(firstAccount.address)).to.equal(secondBorrowLimit)
+    it('emits a proper event', async () => {
+      await expect(oracle.connect(manager).setMaxBorrowerLimit(borrower.address, 1_000_000))
+        .to.emit(oracle, 'MaxBorrowerLimitChanged')
+        .withArgs(borrower.address, 1_000_000)
     })
   })
 
-  describe('ineligibility', () => {
-    const firstScore = 100
-    const secondScore = 200
+  describe('setEligibleForDuration', () => {
+    it('only owner can set eligible for duration', async () => {
+      await expect(oracle.connect(manager).setEligibleForDuration(borrower.address, 10 * DAY))
+        .to.be.revertedWith('Ownable: caller is not the owner')
+      await expect(oracle.connect(borrower).setEligibleForDuration(borrower.address, 10 * DAY))
+        .to.be.revertedWith('Ownable: caller is not the owner')
+    })
 
-    beforeEach(async () => {
-      await oracle.connect(manager).setScore(firstAccount.address, firstScore)
-      await oracle.connect(manager).setScore(secondAccount.address, secondScore)
-      await oracle.connect(owner).setIneligible(firstAccount.address, true)
-      await oracle.connect(owner).setOnHold(firstAccount.address, true)
+    it('eligibility is properly set', async () => {
+      const tx = await oracle.setEligibleForDuration(borrower.address, 10 * DAY)
+      const timestamp = BigNumber.from((await provider.getBlock(tx.blockNumber)).timestamp)
+      expect(await oracle.eligibleUntilTime(borrower.address)).to.equal(timestamp.add(10 * DAY))
+      expect(await oracle.status(borrower.address)).to.equal(Status.Eligible)
     })
 
-    it('ineligibility and on hold set correctly', async () => {
-      expect(await oracle.onHold(firstAccount.address)).to.be.true
-      expect(await oracle.ineligible(firstAccount.address)).to.be.true
+    it('emits a proper event', async () => {
+      await expect(oracle.setEligibleForDuration(borrower.address, 10 * DAY))
+        .to.emit(oracle, 'EligibleUntilTimeChanged')
```

Comment:
We could also check `.withArgs()` here.
There is a trick to do it for time dependant storage:
`
const tx = oracle.setEligibleForDuration(borrower.address, 10 * DAY)
expect(tx).to.emit(oracle, 'EligibleUntilTimeChanged').withArgs(await oracle.status(borrower.address))
`

Reply:
Whoa this is really cool! Thanks for the tip, I hadn't known we could do tricks like this.